tests/main/listing/task.yaml

```diff
@@ -9,27 +9,26 @@ execute: |
     echo "List prints core snap version"
     if [ "$SPREAD_BACKEND" = "linode" -o "$SPREAD_BACKEND" == "qemu" ] && [ "$SPREAD_SYSTEM" = "ubuntu-core-16-64" ]; then
         echo "With customized images the ubuntu-core snap is sideloaded"
-        expected="^core +.*? +(\d{2}\-\d+) + x\d+ +- *"
+        expected="^core +.* +([0-9]{2}\-[0-9]+) + x[0-9]+ +- *$"
     else
-        expected="^core +.*? +(\d{2}\-\d+) + \d+ +canonical +- *"
+        expected="^core +.* +([0-9]{2}\-[0-9]+) + [0-9]+ +canonical +- *$"
     fi
-    snap list | grep -Pq "$expected"
+    snap list | MATCH "$expected"
 
     echo "List prints installed snap version"
-    expected="^test-snapd-tools +(\\d+)(\\.\\d+)* +x[0-9]+ +-"
-    snap list | grep -Pq "$expected"
+    snap list | MATCH '^test-snapd-tools +[0-9]+(\.[0-9]+)* +x[0-9]+ +- *$'
 
     echo "Install test-snapd-tools again"
     . $TESTSLIB/snaps.sh
     install_local test-snapd-tools
     echo "And run snap list --all"
     output=$(snap list --all |grep test-snapd-tools)
-    if [ $(printf "$output" | grep test-snapd-tools | wc -l) != "2" ]; then
+    if [ "$(printf "$output" | grep -c test-snapd-tools)" != "2" ]; then
         echo "Expected two test-snapd-tools in the output, got:"
         echo $output
         exit 1
     fi
-    if [ $(printf "$output" | grep disabled | wc -l) != "1" ]; then
+    if [ "$(printf "$output" | grep -c disabled)" != "1" ]; then
```

Comment:
eh, didn't want to change it that much, but you're right, the use of printf there is weird.